### DISCUSSION TITLE
Traffic video detection at nighttime: A look at why accuracy is key

### DISCUSSION THREAD

**Discussion by NLadi** [ultralytics/ultralytics#22284] [Show and tell]
Nighttime traffic is a whole different challenge.

In her latest blog, Abirami Vina dives into why accuracy is critical when detecting vehicles in low-light conditions, and how computer vision can help make our roads safer and smarter. 🚦

Learn more ➡️ https://www.ultralytics.com/blog/traffic-video-detection-at-nighttime-a-look-at-why-accuracy-is-key

[![long-exposure-shot-of-night-traffic-on-a-freeway-n-2025-02-02-20-06-34-utc](https://github.com/user-attachments/assets/8525e993-3a65-4bc8-9815-e76a37a43240)](https://www.ultralytics.com/blog/traffic-video-detection-at-nighttime-a-look-at-why-accuracy-is-key)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @NLadi, thanks for starting the discussion and sharing insights on nighttime traffic detection 🚀 This is an automated response to help you get quick guidance while an Ultralytics engineer reviews and assists you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Thanks for sharing, @NLadi—if you’re implementing this, fine-tune Ultralytics YOLO11 on low-light data and then run multi-object tracking for stable IDs; for a quick start via CLI: 

```bash
yolo train model=yolo11n.pt data=night_traffic.yaml imgsz=960 epochs=50
yolo track model=runs/detect/train/weights/best.pt source=night.mp4 tracker=bytetrack.yaml save=True
```

For dataset structure see the [datasets guide](https://docs.ultralytics.com/datasets/), tracking usage is covered in [Track mode docs](https://docs.ultralytics.com/modes/track/), and for traffic-specific workflows see [Optimize Traffic with YOLO11](https://www.ultralytics.com/blog/optimizingtraffic-management-with-ultralytics-yolo11) and [YOLO11 AI for Safer Roads](https://www.ultralytics.com/blog/road-safety-with-ultralytics-yolo11-ai-detection-for-safer-streets).
